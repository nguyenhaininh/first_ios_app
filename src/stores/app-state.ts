/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
export interface IAppState {
  informations?: IAppInformation[];
  isShowingMenu: boolean;
  mode: AppMode;
  tab: AppTab;
}

export enum AppMode {
  Init,
  Login,
  Member,
}

export enum AppTab {
  Feed,
  Ranking,
  Favorite,
  Download,
  Search
}

export interface IAppInformation {
  infoId: string;
  subject: string;
  description: string;
  date: string;
  important?: boolean;
}

export function createInitialAppState(): IAppState {
  return {
    isShowingMenu: false,
    mode: AppMode.Init,
    tab: AppTab.Feed,
  };
}
