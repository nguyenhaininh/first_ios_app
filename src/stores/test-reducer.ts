/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import { incrementAction, decrementAction, resetAction } from '../actions/test-actions';

export interface ITestState {
  counter: number;
  message?: string;
}

export function createInitialTestState(): ITestState {
  return { counter: 0 };
}

export const incrementReducer = incrementAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
    };
  }
  return {
    message: undefined,
    counter: state.counter += action.payload,
  };
});

export const decrementReducer = decrementAction.reducer<ITestState>((state, action) => ({
  counter: state.counter -= action.payload,
}));

export const resetReducer = resetAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
      counter: 0
    };
  }
  return {
    message: undefined,
    counter: 0
  };
});

export const testReducer = createTypeReducer(
  createInitialTestState,
  incrementReducer,
  decrementReducer,
  resetReducer,
);
