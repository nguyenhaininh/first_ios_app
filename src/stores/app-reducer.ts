/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import {
  initAppAction, fetchInfomartionsAction, setShowAppMenu, setTabAction, setModeAction,
} from '../actions/app-actions';
import { IAppState, createInitialAppState, AppMode } from './app-state';

export const appReducer = createTypeReducer<IAppState>(
  createInitialAppState,
  initAppAction.reducer((state, action) => {
    if (isError(action)) {
      throw action.error;
    }
    return {
      ...state,
      mode: AppMode.Login,
    };
  }),
  fetchInfomartionsAction.reducer((state, action) => isError(action) ? ({}) : ({
    informations: action.payload
  })),
  setShowAppMenu.reducer((state, action) => ({
    isShowingMenu: action.payload
  })),
  setTabAction.reducer((state, action) => ({
    tab: action.payload
  })),
  setModeAction.reducer((state, action) => ({
    mode: action.payload
  })),
);
