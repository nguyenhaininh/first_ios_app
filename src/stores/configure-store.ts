/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createStore, combineReducers, applyMiddleware } from 'redux';

import { typePendingReducerSet, TypeReduxPendingState, createTypeReduxInitialState, typeReduxMiddleware } from '../common/utility/redux';

import { ITestState, testReducer, createInitialTestState } from './test-reducer';
import { appReducer } from './app-reducer';
import { createInitialAppState, IAppState } from './app-state';

export interface IStore extends TypeReduxPendingState {
  testState: ITestState;
  appState: IAppState;
}

export const rootReducer = combineReducers<IStore>({
  ...typePendingReducerSet,
  testState: testReducer,
  appState: appReducer,
});

const middlewares = [typeReduxMiddleware];

export const InitialState: IStore = {
  ...createTypeReduxInitialState(),
  testState: createInitialTestState(),
  appState: createInitialAppState(),
};

export const configureStore = (initialState: IStore = InitialState) => createStore(rootReducer, initialState, applyMiddleware(...middlewares));
