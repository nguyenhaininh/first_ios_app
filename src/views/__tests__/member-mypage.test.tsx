/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import { Provider } from 'react-redux';
import MemberMyPage from '../member-mypage';

import * as renderer from 'react-test-renderer';
import { configureStore } from '../../stores/configure-store';

it('renders correctly', () => {
  const store = configureStore();
  const tree = renderer.create(
    <Provider store={store}>
      <MemberMyPage />
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
