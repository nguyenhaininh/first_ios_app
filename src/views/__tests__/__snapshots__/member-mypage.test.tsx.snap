// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
    }
  }
  style={
    Object {
      "backgroundColor": "#f7f7f7",
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4f5b77",
            "flexDirection": "row",
            "height": 42,
            "marginTop": 20,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          [missing "en.account_info" translation]
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.change_email" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.change_payment" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.contact_detail" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.point_setting" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.billing_history" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.stop_payment" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4f5b77",
            "flexDirection": "row",
            "height": 42,
            "marginTop": 20,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          [missing "en.various_setting" translation]
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.download_wifi_only" translation]
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "bottom": 3,
                "color": "#333333",
                "fontSize": 12,
                "left": 15,
                "position": "absolute",
              }
            }
          >
            [missing "en.tap_to_view_explain" translation]
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.delivery_setting" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.confirm_user_terminate" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4f5b77",
            "flexDirection": "row",
            "height": 42,
            "marginTop": 20,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          [missing "en.other_setting" translation]
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.contact_us" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.help_faq" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.company_profile" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.term_of_service" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.personal_info" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.privacy_policy" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#eaebed",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 54,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            [missing "en.logout" translation]
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/arrow-ico.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
